snuba/web/views.py

```diff
@@ -373,22 +375,19 @@ def sdk_distribution(*, timer: Timer) -> Response:
     return format_result(run_query(dataset, request, timer))
 
 
-@application.route("/subscriptions", methods=["POST"])
-def create_subscription():
+@application.route("/<dataset:dataset>/subscriptions", methods=["POST"])
+def create_subscription(*, dataset: Dataset):
     return (
         json.dumps({"subscription_id": uuid1().hex}),
         202,
         {"Content-Type": "application/json"},
     )
 
 
-@application.route("/subscriptions/<uuid>/renew", methods=["POST"])
-def renew_subscription(uuid):
-    return "ok", 202, {"Content-Type": "text/plain"}
-
-
-@application.route("/subscriptions/<uuid>", methods=["DELETE"])
-def delete_subscription(uuid):
+@application.route(
+    "/<dataset:dataset>/subscriptions/<int:partition>/<key>", methods=["DELETE"]
```

Comment:
I wanted to use the UUID validator here or `key` but it assumes hyphenation. 😞

Reply:
Does it make sense to just have `key` here rather than `partition/key`? That way we can just return a key like `1:<uuid>` and parse it on our end

Reply:
We'll have to parse it at some point, though. I guess there are a couple things we have to decide:

1. what separator to use (I'm ok with `/` personally),
2. where we want to validate and split the key (I would think we need the partition index here to figure out what partition this record needs to be deleted from, but I'm only thinking about a hypothetical implementation at this point)

Reply:
I suppose using the `/` as the separator is fine. I can just not escape the id when passing it in the url from sentry, and then from sentry's point of view there's just a single key, and we treat it more like a path.

Reply:
Yeah, this is kind of what I figured — Sentry can treat the `partition/uuid` as a single identifier (or even the whole path or URL as the identifier.)